<launch>
    <arg name="world" default="office"/>
    <arg name="method" default="heuristic_uncertainty"/>
    <arg name="inference" default="pessimistic"/>
    <arg name="nrobots" default="2"/>
    <arg name="nareas" default="4"/>
    <arg name="decay" default="non_uniform"/>
    <arg name="dsteps" default="4"/>
    <arg name="tframe" default="100"/>
    <arg name="placement" default="1"/>
    <arg name="fileposes" default="office_n4_sampled_nodes_poses_dict"/>
    <arg name="fileresult" default="office_n8_treebased_decision_oracle"/>
    <arg name="save" default="False"/>

    <param name="/use_sim_time" value="true" />
    <rosparam command="load" file="$(find int_preservation)/param/mission.yaml" subst_value="true"/>

    <node name="Stage" pkg="stage_ros" type="stageros" args="$(find int_preservation)/world/$(arg world)$(arg nrobots).world">
        <param name="base_watchdog_timeout" value="0" />
    </node>

    <node name="map_server" pkg="map_server" type="map_server" args="$(find int_preservation)/param/$(arg world)_map.yaml">
        <param name="frame_id" value="map" />
    </node>

    <include file="$(find int_preservation)/launch/robots/robot_sim$(arg nrobots).launch">
        <arg name="method" value="$(arg method)"/>
        <arg name="world" value="$(arg world)"/>
    </include>

    <include file="$(find int_preservation)/launch/areas/area_sim$(arg nareas).launch">
        <arg name="decay" value="$(arg decay)"/>
    </include>

    <!-- RVIZ to view the visualization -->
    <node name="RVIZ" pkg="rviz" type="rviz" args=" -d $(find int_preservation)/param/mission.rviz" output="screen" />

</launch>
